pkg/network/dns_snooper_test.go

```diff
@@ -166,28 +191,46 @@ func sendDNSQuery(
 	}
 
 	dnsClient := mdns.Client{Net: strings.ToLower(protocol.String()), Dialer: localAddrDialer}
-
 	dnsHost := net.JoinHostPort(serverIP, "53")
-	rep, _, _ := dnsClient.Exchange(msg, dnsHost)
-	return queryIP, queryPort, rep
+	var reps []*mdns.Msg
+
+	if protocol == TCP {
+		conn, err := dnsClient.Dial(dnsHost)
+		require.NoError(t, err)
+		for _, domain := range domains {
+			msg.SetQuestion(mdns.Fqdn(domain), mdns.TypeA)
+			rep, _, _ := dnsClient.ExchangeWithConn(msg, conn)
+			reps = append(reps, rep)
+		}
+	} else { // UDP
+		for _, domain := range domains {
+			// msg.SetQuestion(mdns.Fqdn(domains[0]), mdns.TypeA)
```

Comment:
Is this to be deleted, or just commented out while you change something?

Reply:
Good catch. Will delete it.